File: solutions/go/perfect-numbers/1/perfect_numbers.go
package perfect

import "errors"

type Classification int

const (
	ClassificationDeficient Classification = iota
	ClassificationPerfect
	ClassificationAbundant
)

var ErrOnlyPositive = errors.New("given number was not positive")

func Classify(n int64) (Classification, error) {
	if n <= 0 {
		return 0, ErrOnlyPositive
	}

	var sum int64
	for i := int64(1); i <= n/2; i++ {
		if n%i == 0 {
			sum += i
		}
	}

	if sum == n {
		return ClassificationPerfect, nil
	} else if sum > n {
		return ClassificationAbundant, nil
	}
	return ClassificationDeficient, nil
}

File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    """A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")
    match sum(num for num in range(1, number // 2 + 1) if number % num == 0):
        case s if s == number: return "perfect"
        case s if s > number:  return "abundant"
        case _:                return "deficient"
